src/main/java/org/apache/sling/event/impl/jobs/stats/GaugeSupport.java

```diff
@@ -0,0 +1,128 @@
+     * @param metricRegistry the (sling) metric registry */
+    GaugeSupport(final Statistics globalQueueStats, MetricRegistry metricRegistry) {
+        this(null, globalQueueStats, metricRegistry);
+    }
+
+    /** Creates a new GaugeSupport instance. Registers gauges for jobs (based on the queueStats).
+     *
+     * @param queueName name of the queue
+     * @param queueStats queueStats of that queue
+     * @param metricRegistry the (sling) metric registry */
+    GaugeSupport(String queueName, final Statistics queueStats, MetricRegistry metricRegistry) {
+        this.metricRegistry = metricRegistry;
+        if (metricRegistry != null) {
+            gaugeList.put(getMetricName(queueName, FINISHED_METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getNumberOfFinishedJobs();
+                }
+            });
+            gaugeList.put(getMetricName(queueName, CANCELLED_METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getNumberOfCancelledJobs();
+                }
+            });
+            gaugeList.put(getMetricName(queueName, FAILED__METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getNumberOfFailedJobs();
+                }
+            });
+            gaugeList.put(getMetricName(queueName, QUEUED_METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getNumberOfQueuedJobs();
+                }
+            });
+            gaugeList.put(getMetricName(queueName, PROCESSED_METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getNumberOfProcessedJobs();
+                }
+            });
+            gaugeList.put(getMetricName(queueName, ACTIVE_METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getNumberOfActiveJobs();
+                }
+            });
+            gaugeList.put(getMetricName(queueName, AVG_WAITING_TIME_METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getAverageWaitingTime();
+                }
+            });
+            gaugeList.put(getMetricName(queueName, AVG_PROCESSING_TIME_METRIC_SUFFIX), new Gauge() {
+
+                public Object getValue() {
+                    return queueStats.getAverageProcessingTime();
+                }
+            });
+            for (Map.Entry<String, Gauge> entry : gaugeList.entrySet()) {
+                metricRegistry.register(entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+    private String getMetricName(String queueName, String metricSuffix) {
+        return GAUGE_NAME_PREFIX + (queueName != null ? "." + QUEUE_PREFIX + "." + queueName : "") + metricSuffix;
```

Comment:
Hi Stefan, thanks a lot for the comprehensive review (I'm currently working on the changes). I was indeed considering the sanitization of name but realized that special chars are fine in Metrics itself and should be sanitized when further use the names, e.g. in reporters. But I'm fine with adding it here!

Reply:
I think having sanitization here would be preferrable as it might make life easier for downstream usages of metrics. Re the sanitization I was thinking of a combination of everything lowercase + every special character (non alphanumeric) replaced with `_`. Which then I guess this could result - in theory - in name clashes, so that's maybe something to double check and then if it happens perhaps adding something like `_<counter>` ..
wdyt?